jangaroo-browser/src/main/joo/js/DataTransferItemList.as

```diff
@@ -0,0 +1,40 @@
+package js {
+
+/**
+ * A list of DataTransferItem objects representing items being dragged. During a drag operation, each DragEvent has a
+ * dataTransfer property and that property is a DataTransferItemList.
+ */
+[Native("DataTransferItemList")]
+public class DataTransferItemList {
+  /**
+   * Returns the number of items in the drag data store.
+   */
+  public native function get length(): Number;
+
+  //noinspection ReservedWordAsName
+  /**
+   * Adds a new entry for the given data to the drag data store. If the data is plain text then a type string has to be provided also.
+   *
+   * @param data can be a String or {@link File}
+   * @param type If the data is plain text then a type string has to be provided also.
+   */
+  public native function add(data: *, type: String = undefined): DataTransferItem;
+
+  /**
+   * Removes all the entries in the drag data store.
+   */
+  public native function clear(): void;
+
+  public native function item(index: Number): DataTransferItem;
+
+  /**
+   * Removes the indexth entry in the drag data store.
+   */
+  public native function remove(index: Number): void;
+
+  /*
+  Cannot be expressed in ActionScript:
+  [name: Number]: DataTransferItem;
```

Comment:
Does not matter, we can use `item(index)` for typed access.
Or check whether we can claim to `extend Array` and use `ArrayElementType` annotation at class level - is it allowed? Does IDEA support this?

Reply:
Tried that but It does not have all Array methods so this would also not be good